### DISCUSSION TITLE
Linux App with multiple (thousands of ) client connections.

### DISCUSSION THREAD

**Discussion by Yashavanth-CP** [uNetworking/uWebSockets#1555] [General]
I am building an application, that runs on linux OS.  Basically, its a Backend application that creates a new websocket client connection to the external server as soon as the user login to the application. As of now, the number of users are in the range of 28k to 30k.  I would like to explore the possibility of using the uWebSockets library for achieving this.

Couple of questions on the same...
1. Is there already a test suite which tests this scenario ?
2. Conceptually, is the requirement is achievable as per the library design ?

It would be of great help, if I get a lead on this to get started. I have done basic research with uWebsocket with one client till now.

Note : Client connections are dynamic (added and deleted dynamically in any order).

Best Regards,
Yash

**Comment by uNetworkingAB** [comment]
As long as you are aware uWS only does the server part, not client part, yes then it's designed for this. There is no automated testsuite but there is a benchmark called scale_test or something like that in benchmarks folder where you can test a million websockets or so
